test/ducks/apps/components/applicationPage/details.spec.js

```diff
@@ -67,4 +73,101 @@ describe('ApplicationPage details component', () => {
     const component = shallow(<Details {...appProps} />).getElement()
     expect(component).toMatchSnapshot()
   })
+
+  it('should handle channel switching if with installed app from registry provided', async () => {
+    jest.doMock('ducks/apps', () => ({
+      REGISTRY_CHANNELS,
+      getContext: () =>
+        Promise.resolve({
+          attributes: {
+            debug: true
+          }
+        })
+    }))
+    // we have to re import to handle previous mocking here
+    const MockedDetails = require('ducks/apps/components/ApplicationPage/Details')
+      .Details
+    const appProps = getAppProps()
+    // correctly handled if from registry and installed
+    appProps.app.installed = true
+    appProps.app.source = `registry://${appProps}/stable`
+    const component = shallow(<MockedDetails {...appProps} />)
+    expect(component.state('displayBetaChannel')).toBe(false)
```

Comment:
I think we should not test the implementation (i.e. state attributes).
Asserting the snapshot is correct once the channels have been toggled is enough.

Moreover it's simpler to keep the test consistent when the implementation changes but not the final rendering.